app/src/main/java/swati4star/createpdf/util/PDFUtils.java

```diff
@@ -578,34 +578,53 @@ public boolean reorderRemovePDF(String inputPath, String output, String pages) {
         return outputPaths;
     }
 
+    /**
+     * checks if the user entered split ranges are valid or not
+     *
+     * @param path   the input pdf path
+     * @param ranges string array that contain page range, can be a single integer or range separated by dash like 2-5
+     * @return true if input is valid, otherwise false
+     */
     private boolean isInputValid(String path, String[] ranges) {
         try {
             PdfReader reader = new PdfReader(path);
             int numOfPages = reader.getNumberOfPages();
-            int startPage;
-            int endPage;
+            return checkRangeValidity(numOfPages, ranges);
```

Comment:
instead of returning it do it like this.

```Java
int result = checkRangeValidity(numOfPages, ranges);
switch(result) {
    case ERROR_PAGE_NUMBER:
         show snackbar and return false
   case ERROR_PAGE_RANGE:
         show snackbar and return false
   default:
         return true;
}
```